### DISCUSSION TITLE
Check this out: running web assembly from postgres

### DISCUSSION THREAD

**Discussion by haikyuu** [geldata/gel#2109] [Ideas]
## Rust example
```rust
#[no_mangle]
pub extern fn sum(x: i32, y: i32) -> i32 {
    x + y
}
```
Compile it to web assembly and then
```sql
-- New instance of the `simple.wasm` WebAssembly module.
SELECT wasm_new_instance('/absolute/path/to/simple.wasm', 'ns');

-- Call a WebAssembly exported function!
SELECT ns_sum(1, 2);

--  ns_sum
-- --------
--       3
-- (1 row)
```

And the performance is amazing:

Benchmark | Runtime | Time (ms) | Ratio
-- | -- | -- | --
Fibonacci (n = 50) | postgres-ext-wasm | 0.206 | 1×
PL/pgSQL | 0.431 | 2×
Fibonacci (n = 500) | postgres-ext-wasm | 0.217 | 1×
PL/pgSQL | 2.189 | 10×
Fibonacci (n = 5000) | postgres-ext-wasm | 0.257 | 1×
PL/pgSQL | 18.643 | 73×

Check it out here:
https://github.com/wasmerio/wasmer-postgres

This could be a great way to extend EdgeDB as well !

**Comment by 1st1** [comment]
Wow, nice! 

> This could be a great way to extend EdgeDB as well !

We have some vague ideas about making an official EdgeDB extension API use wasm, but nothing concrete yet. Probably will be one of the focus areas after 1.0 release.
